### ISSUE TITLE
Error opening remote file at https://db4s.dbhub.io/. The root CA certificate is not trusted for this purpose

### ISSUE THREAD

**Issue by gvoc** [sqlitebrowser/sqlitebrowser#2829]
Error opening remote file at https://db4s.dbhub.io/.
The root CA certificate is not trusted for this purpose

--------------------

#### What did you do?
I selected my identity

#### What did you expect to see?
I expected all my DBs to be loaded

#### What did you see instead?
Error opening remote file at https://db4s.dbhub.io/.
The root CA certificate is not trusted for this purpose

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [ ] Windows: ( _version:_ ___ )
- [ ] Linux: ( _distro:_ ___ )
- [X] macOS: Big Sur version:11.5.2 (20G95)
- [ ] Other: ___

#### What is your DB4S version?

- [X] 3.12.2
- [ ] 3.12.1
- [ ] 3.12.0
- [ ] 3.11.x
- [ ] Other: ___

#### Did you also

- [ ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [X] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Oh, that's interesting.  That error message, that comes from DB4S yeah?

Asking because we've occasionally seen people try going to that website with other programs (for example, a web browser) which won't understand the certificate as it's not a web browser one.

**Comment by gvoc** [comment]
Yes, it comes from DB4S.  I tested it on WindowsOS and it's running normal.  But not in Big Sur.

**Comment by justinclift** [comment]
Interesting.  Sounds like the macOS keychain might be getting in the way somehow.

I'll have to take a look at it later on a Big Sur box.  Thanks for letting us know. :smile:

**Comment by ARSBento** [comment]
Having the same issue on Big Sur

**Comment by SuperKLA** [comment]
Same issue here, I am using Big Sur

**Comment by justinclift** [comment]
It _sounds_ like Big Sur has some weird configuration problem, that isn't passing DB4S the right certificate.

I think the only way we'll be able to diagnose this, is to install Big Sur on a mac that we have access to, then trigger the problem while it's running in a debugger to find out exactly WTF is going on. :frowning:

**Comment by SuperKLA** [comment]
@justinclift is there any chance we see a fix for that soon?

**Comment by justinclift** [comment]
k, it happens in macOS Monterey (12.5.1) as well.  As a rough guess, it seems like a problem from Big Sur (11.6.8) onwards.

**Comment by chrisjlocke** [comment]
Thanks for not only reporting the issues, but suggesting fixes.  Useful and appreciated. 👍

**Comment by justinclift** [comment]
> The best way is to switch to a proper CA, e.g., Let’s Encrypt.

Thanks.  We have our own "self-signed" CA (root, intermediate, etc) which we use internally for things.  A CA with the root outside of our control wouldn't meet our needs.

Or maybe I'm misunderstanding, as this bit ...:

> Server-side authentication can still use DBHub CA.

... confuses me.  You're saying we could do both at the same time in some way? :confused: :wink:

I'll take a look at those options you suggest too.  They look worth investigating.  I've not touched GnuTLS before, instead using (as you guessed) OpenSSL.  Heh Heh Heh. :wink:

**Comment by justinclift** [comment]
Leaving this here as reference info, for when I pick this up to investigate again:

https://en.wikipedia.org/wiki/Rpath
